### ISSUE TITLE
Incorrectly handling basemap's old -D option?

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#6694]
**Description of the problem**

I'm running an old script, which uses `basemap`'s **-D** option. The **-D** option is no longer available in GMT's modern mode, so I expect GMT should handle it in a backward-compatible way, or should raise an error. Instead, GMT just skips the `basemap` command silently.

Here is the minimal script to reproduce the issue:
```
gmt begin map
gmt basemap -R-10/10/-10/10 -JX10c -Baf -BWSne
gmt basemap -DjTR+w2.9i/3.9i+o0.05i -F+p+ggray95
gmt end show
```

If you add `-Vd` to the 2nd `basemap` call, you will see a compatibility warning:
```
basemap [COMPATIBILITY]: Option -D is not available in modern mode - see inset instead
```
If backward compatibility is impossible or is difficult, we should raise an error.

**System information**

* Operating system: Linux
* GMT version (`gmt --version`): `6.4.0_fab9a09_2022.05.05`

**Comment by PaulWessel** [comment]
I think perhaps an error is better.  It seems too much work to make this happen seamlessly.  There is no guarantee that modern mode scripts be fully compatible with a classic option - after all we changed lots going to modern mode and there are no old modern mode scripts we must remain compatible with (except from going forward).

So I will add a PR where we escalate **-D** under modern mode to an error.

**Comment by seisman** [comment]
Closed by #6696.
